### PR TITLE
build: add github action ci

### LINKED ISSUE #841: Issue when npm install
Hi,
Having an issue on `npm install` after cloning.

branch : **master**
last sha: 33f9f12db20a2b1000cd3b0f79cbf4dae2397627
command run : `npm install`
context: **personnal** (not axa)
os: **windows 10**
node version: **14.15.1**
npm version: **7.0.15**

error : 
image

### PR DESCRIPTION
**Pull request by jforatier**
## Related issue
Fixes #841 

### Reference to the issue

Not fix but GH-Action to check

### Description of the issue

`A GitHub action implementation to validate/check npm install / npm test (on push and PR)`

### Person(s) for reviewing proposed changes

`You can add @mention here`

# Important

Before creating a pull request run unit tests

```sh
$ npm test

# watch for changes
$ npm test -- --watch

# For a specific file (e.g., in packages/context/__tests__/command.test.js)
$ npm test -- --watch packages/action
```

### CODE REVIEW

**Comment by arnaudforaison** [comment]
Bonne initiative, je pense qu'il faut intégrer ça. Etude en cours ;-)

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+name: Toolkit - Build & Test
+
+on: [push, pull_request]
```
Comment (arnaudforaison) [inline .github/workflows/ci.yml:3]:
```suggestion
on: 
  push:
    branches:
      - master
      - release/**
  pull_request:
    branches:
      - master
      - release/**
```

**Review by arnaudforaison — CHANGES_REQUESTED** [review]

**Review by arnaudforaison — APPROVED** [review]

**Review by bernard-arnaud — APPROVED** [review]

**Review by LunarzFR — DISMISSED** [review]

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+    steps:
+      - uses: actions/setup-node@v2
+        with:
+          node-version: ${{ matrix.node }}
+
+      - name: checkout
+        uses: actions/checkout@v2
+
+      - run: npm install
```
Comment (arnaudforaison) [inline .github/workflows/ci.yml:30]: npm ci

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
+name: Toolkit - Build & Test
```
Comment (arnaudforaison) [inline .github/workflows/ci.yml:1]: Changer le nom en `Compatibility npm on multi OS & Node version` + renommer le fichier en `compatibility-npm.yml`

**Review by arnaudforaison — CHANGES_REQUESTED** [review]

**Review by arnaudforaison — APPROVED** [review]

**Pull request merged**